lua/wire/server/wirelib.lua

```diff
@@ -1217,6 +1217,20 @@ else
 	end
 end
 
+if CPPI and FindMetaTable("Entity").CPPICanDamage then
+	---@param player Player
+	---@param target Player
+	function WireLib.CanDamage(player, target)
+		return target:CPPICanDamage(player)
+	end
+else
+	---@param player Player
+	---@param target Player
+	function WireLib.CanDamage(player, target)
+		return hook.Run("PlayerShouldTakeDamage", target, player)
```

Comment:
Not sure if this hook is correct. Target is expected to be a player I think.

Reply:
Yeah I made this WireLib function intending for only players to be passed but I'm using it in E2 which allows passing non-player Entities..

Might need to switch to https://wiki.facepunch.com/gmod/GM:EntityTakeDamage but pass a dummy 0 damage struct or something..

Reply:
I don't think that's a good solution. Probably better to go with Wirelib.CanTool.